### ISSUE TITLE
select from dataframe fails when array element contains a nested object

### ISSUE THREAD

**Issue by nimiby** [elastic/elasticsearch-hadoop#729]
Consider the following document structure:
```
{
  "key": "val",
  "arr_key": [
    {
      "item": 1,
      "obj_key": {
        "key_1": "val1",
        "key_2": "val2"
      }
    },
    {
      "item": 2,
      "obj_key": {
        "key_1": "val3",
        "key_2": "val4"
      }
    }
  ]
}
```

"arr_key" is marked to be an array property using:
es.read.field.as.array.include=arr_key

When trying to load this document as dataframe (using esDF()) it parses the following schema from the mapping (successfully):
```
   root
    |-- arr_key: array (nullable = true)
    |    |-- element: struct (containsNull = true)
    |    |    |-- item: long (nullable = true)
    |    |    |-- obj_key: struct (nullable = true)
    |    |    |    |-- key_1: string (nullable = true)
    |    |    |    |-- key_2: string (nullable = true)
    |-- key: string (nullable = true)
```

but try to select the dataframe produces the following error:
```
org.elasticsearch.hadoop.EsHadoopIllegalStateException: Position for 'arr_key.obj_key.key_2' not found in row; typically this is caused by a mapping inconsistency
	at org.elasticsearch.spark.sql.RowValueReader$class.addToBuffer(RowValueReader.scala:45)
	at org.elasticsearch.spark.sql.ScalaRowValueReader.addToBuffer(ScalaEsRowValueReader.scala:14)
	at org.elasticsearch.spark.sql.ScalaRowValueReader.addToMap(ScalaEsRowValueReader.scala:84)
	at org.elasticsearch.hadoop.serialization.ScrollReader.map(ScrollReader.java:794)
	at org.elasticsearch.hadoop.serialization.ScrollReader.read(ScrollReader.java:692)
	at org.elasticsearch.hadoop.serialization.ScrollReader.map(ScrollReader.java:794)
	at org.elasticsearch.hadoop.serialization.ScrollReader.read(ScrollReader.java:692)
	at org.elasticsearch.hadoop.serialization.ScrollReader.list(ScrollReader.java:734)
	at org.elasticsearch.hadoop.serialization.ScrollReader.read(ScrollReader.java:687)
	at org.elasticsearch.hadoop.serialization.ScrollReader.map(ScrollReader.java:794)
	at org.elasticsearch.hadoop.serialization.ScrollReader.read(ScrollReader.java:692)
	at org.elasticsearch.hadoop.serialization.ScrollReader.readHitAsMap(ScrollReader.java:457)
	at org.elasticsearch.hadoop.serialization.ScrollReader.readHit(ScrollReader.java:382)
	at org.elasticsearch.hadoop.serialization.ScrollReader.read(ScrollReader.java:277)
	at org.elasticsearch.hadoop.serialization.ScrollReader.read(ScrollReader.java:250)
	at org.elasticsearch.hadoop.rest.RestRepository.scroll(RestRepository.java:456)
	at org.elasticsearch.hadoop.rest.ScrollQuery.hasNext(ScrollQuery.java:86)
	at org.elasticsearch.spark.rdd.AbstractEsRDDIterator.hasNext(AbstractEsRDDIterator.scala:43)
```

seems like whenever there is a nested object structure under an array element, it fails to read the values from the nested properties of the object.

tested with version 2.2.0

**Comment by costin** [comment]
Pushed a fix in master. Will be backported to 2.2

**Comment by DrSpez** [comment]
@rjof did you fix it? I get the same error in 5.4.3

**Comment by bytemedwb** [comment]
I am also getting this error in 5.4.   Is there a fix?
I am unable to move to a newer elastic version.
